### ISSUE TITLE
Scaling with docker doesn't appear to work

### ISSUE THREAD

**Issue by stahnma** [yyyar/gobetween#131]
I am using a fairly straightforward docker-compose file. When I scale down, backends dropout, healthcheck fails, and thus routing stops. When I scale up however, using docker-compose, it doesn't seem like gobetween ever notices. It doesn't start sending healthchecks to the second instance, unless I restart gobteween. 

I haven't found a way to HUP gobtween or use the API to rescan. Is there something I'm missing? 

Gist of my toml config: https://gist.github.com/stahnma/0e8b4dff875b102201af071a0991990f

**Comment by nickdoikov** [comment]
hello @stahnma , sorry but i do not catch your idea . I mean : ``` It doesn't start sending healthchecks to the second instance, unless I restart gobteween.```

please clarify .
gb do not send any heartbeat at all. 
also there is no sync mechanism between gb instances for now , but you can use consul for external checks as well as for storing initial config for gb .

**Comment by stahnma** [comment]
Ok, let me try again.

Basically, I am trying to scale out puppet servers (just https). I have gb looking at docker for discovery. When I do `docker-compose scale puppet=2` gb never starts routing anything to the second puppet server, even though it has the same docker labels, etc. I would expect gb to notice the new instance (via docker discovery) and start routing to that as a backend (while using healthchecks to see if the backend is alive or not). Does that make sense?

**Comment by stahnma** [comment]
Maybe I'm confused. How do you re-trigger discovery at all? It looks like it only reads it upon startup. I tried using an `exec` discovery method to see if it was a bug in docker discovery, but maybe I am misunderstanding the software overall. Can gobetween dynamically adjust the number of backends without a restart?

**Comment by stahnma** [comment]
Ok, I figured it out. I was missing an `interval` key in my toml file to tell it to redo discovery periodically. Sorry for my confusion. Closing issue.

**Comment by nickdoikov** [comment]
healthchecks applying to all discovered backends each healthcheks interval time , if healthcheck failed - backend node marked as faulty .  
so , seems like you backend discovery not properly configured , or your checks faulty . 

Please notice that i mean not a docker based checks but gb checks and gb docker(or any other )
 discovery.
Could you please add here your config .
